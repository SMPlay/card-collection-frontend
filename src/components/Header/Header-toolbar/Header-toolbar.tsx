import React from "react";
import { Toolbar, IconButton, Typography } from "@material-ui/core";
import MenuIcon from "@material-ui/icons/Menu";
import clsx from "clsx";
import {
  makeStyles,
  Theme,
  createStyles,
} from "@material-ui/core/styles";

interface HeaderToolbarProps {
  handleDrawerOpen: () => void;
  currentPage: string;
  open: boolean;
}


const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    hide: {
      display: "none",
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
  })
);

export const HeaderToolbar: React.FC<HeaderToolbarProps> = ({ handleDrawerOpen, currentPage, open }) => {
  const classes = useStyles();

  return (
    <Toolbar>
      <IconButton
        color="inherit"
        aria-label="open drawer"
        onClick={handleDrawerOpen}
        edge="start"
        className={clsx(classes.menuButton, open && classes.hide)}
      >
        <MenuIcon />
      </IconButton>
      <Typography variant="h6" noWrap>
        {currentPage}
      </Typography>
    </Toolbar>
  );
};
