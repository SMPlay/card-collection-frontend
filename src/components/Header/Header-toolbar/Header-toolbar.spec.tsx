import React from "react";
import { create } from "react-test-renderer";

import { HeaderToolbar } from "./Header-toolbar";

describe("Header-toolbar component", () => {
  it("should render", () => {
    const props = {
      handleDrawerOpen: jest.fn(),
      currentPage: "test",
      open: true
    }

    const tree = create(<HeaderToolbar {...props}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
