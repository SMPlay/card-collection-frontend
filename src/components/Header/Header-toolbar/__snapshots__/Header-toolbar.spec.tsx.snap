// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header-toolbar component should render 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
>
  <button
    aria-label="open drawer"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 makeStyles-hide-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </span>
  </button>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
  >
    test
  </h6>
</div>
`;
